.github/workflows/templates/workflow-template.yml

```diff
@@ -22,238 +24,43 @@ on:
         type: string
         options:
-        - deploy
-        - destroy
-env:
-  TF_IN_AUTOMATION: true
-  AWS_REGION: "eu-west-2"
-  ENVIRONMENT_MANAGEMENT: ${{ secrets.MODERNISATION_PLATFORM_ENVIRONMENTS }}
+          - deploy
+          - destroy
+
 permissions:
-  id-token: write # This is required for requesting the JWT
-  contents: read # This is required for actions/checkout
-defaults:
-  run:
-    shell: bash
+  id-token: write  # This is required for requesting the JWT
+  contents: read  # This is required for actions/checkout
 
 jobs:
+  strategy:
+    uses: ./.github/workflows/reusable_terraform_strategy.yml
+    if: inputs.action != 'destroy'
+    with:
+      application: "${{ github.workflow }}"
 
-  plan-dev-test:
+  terraform:
+    needs: strategy
+    if: inputs.action != 'destroy'
     strategy:
-      matrix:
-        include:
-          - environment: development
-          - environment: test
-    name: Plan - ${{  matrix.environment }}
-    runs-on: ubuntu-latest
-    if: github.ref != 'refs/heads/main' || github.event_name == 'workflow_dispatch' && github.event.inputs.action == 'deploy'
-    env:
-      TF_ENV: ${{ matrix.environment }}
-    steps:
-      - name: Checkout Repository
-        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
-      - name: Set Account Number
-        run: echo "ACCOUNT_NUMBER=$(jq -r -e --arg account_name "${GITHUB_WORKFLOW}-${TF_ENV}" '.account_ids[$account_name]' <<< $ENVIRONMENT_MANAGEMENT)" >> $GITHUB_ENV
-      - name: configure aws credentials
-        uses: aws-actions/configure-aws-credentials@67fbcbb121271f7775d2e7715933280b06314838 # v1.7.0
-        with:
-          role-to-assume: "arn:aws:iam::${{ env.ACCOUNT_NUMBER }}:role/github-actions"
-          role-session-name: githubactionsrolesession
-          aws-region: ${{ env.AWS_REGION }}
-      - name: Load and Configure Terraform
-        uses: hashicorp/setup-terraform@633666f66e0061ca3b725c73b2ec20cd13a8fdd1 # v2.0.3
-        with:
-          terraform_version: "~1"
-          terraform_wrapper: false
-      - name: Plan - ${{ matrix.environment }}
-        run: |
-          terraform --version
-          echo "Terraform plan - ${TF_ENV}"
-          bash scripts/terraform-init.sh terraform/environments/$GITHUB_WORKFLOW
-          terraform -chdir="terraform/environments/${GITHUB_WORKFLOW}" workspace select "${GITHUB_WORKFLOW}-${TF_ENV}"
-          bash scripts/terraform-plan.sh terraform/environments/$GITHUB_WORKFLOW
+      fail-fast: false
+      matrix: ${{ fromJson(needs.strategy.outputs.matrix) }}
+    uses: ./.github/workflows/reusable_terraform_plan_apply.yml
+    with:
+      application: "${{ github.workflow }}"
+      environment: "${{ matrix.target }}"
+      action: "${{ matrix.action }}"
+    secrets:
+      modernisation_platform_environments: "${{ secrets.MODERNISATION_PLATFORM_ENVIRONMENTS }}"
+      pipeline_github_token: "${{ secrets.DSO_GITHUB_AUTOMATION_PAT }}"
```

Comment:
What is this used for? Should we be using our token here?

Reply:
Good spot, this is for posting comments in a PR (tf plan/apply output) I believe (that's what we used it for in DSO in the past)

Reply:
this PAT is global to organisation, I would be tempted to re-use it (once AL goes away, it would be nice to keep it), but if we have one as well then it makes sense to use ours

Reply:
see our new shiny secret please :)